packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWFInstanceManager.m

```diff
@@ -0,0 +1,100 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTWFInstanceManager.h"
+
+@interface FLTThreadSafeMapTable<KeyType, ObjectType> : NSObject
++ (nonnull FLTThreadSafeMapTable *)strongToStrongMapTable;
+- (void)setObject:(nonnull ObjectType)object forKey:(nonnull KeyType)key;
+- (void)removeObjectForKey:(nonnull KeyType)key;
+- (nullable ObjectType)objectForKey:(nonnull KeyType)key;
+@end
+
+@implementation FLTThreadSafeMapTable {
+  NSMapTable<id, id> *_table;
+  dispatch_queue_t _lockQueue;
+}
+
++ (FLTThreadSafeMapTable *)strongToStrongMapTable {
+  return [[FLTThreadSafeMapTable alloc] initWithTable:[NSMapTable strongToStrongObjectsMapTable]];
+}
+
+- (nonnull instancetype)initWithTable:(NSMapTable *)table {
+  self = [super init];
+  if (self) {
+    _lockQueue = dispatch_queue_create("FLTThreadSafeMapTable", DISPATCH_QUEUE_SERIAL);
+    _table = table;
+  }
+  return self;
+}
+
+- (void)setObject:(nonnull id)object forKey:(nonnull id)key {
+  if (key && object) {
+    dispatch_async(_lockQueue, ^{
+      [self->_table setObject:object forKey:key];
+    });
+  }
+}
+
+- (void)removeObjectForKey:(nonnull id)key {
+  if (key != nil) {
+    dispatch_async(_lockQueue, ^{
+      [self->_table removeObjectForKey:key];
+    });
+  }
+}
+
+- (nullable id)objectForKey:(nonnull id)key {
+  id __block object = nil;
+  dispatch_sync(_lockQueue, ^{
+    object = [self->_table objectForKey:key];
+  });
+  return object;
+}
+@end
+
+@implementation FLTWFInstanceManager {
+  FLTThreadSafeMapTable<NSNumber *, NSObject *> *_instanceIDsToInstances;
+  FLTThreadSafeMapTable<NSObject *, NSNumber *> *_instancesToInstanceIDs;
+}
+
+- (instancetype)init {
+  if (self) {
+    _instanceIDsToInstances = [FLTThreadSafeMapTable strongToStrongMapTable];
+    _instancesToInstanceIDs = [FLTThreadSafeMapTable strongToStrongMapTable];
+  }
+  return self;
+}
+
+- (void)addInstance:(nonnull NSObject *)instance instanceID:(long)instanceID {
+  [_instancesToInstanceIDs setObject:@(instanceID) forKey:instance];
+  [_instanceIDsToInstances setObject:instance forKey:@(instanceID)];
```

Comment:
This (and all the other methods that do multiple operations) are racy if there are ever interactions from multiple threads with the same instance or identifier (depending on the method). E.g., one thread adding an object with an ID and another thread removing with the same ID could end up with this table pair having a mapping in one direction but not the other.

Is there a reason you're doing the serialization at the table operation level instead of this class?